src/lib/Quiz.jsx

```diff
@@ -27,28 +75,35 @@ const Quiz = function ({
 
   useEffect(() => {
     if (shuffle) {
-      setQuestions(shuffleQuestions(quiz.questions));
+      setQuestions([...shuffleQuestions(quiz.questions)]);
     } else {
       quiz.questions.length = nrOfQuestions;
       setQuestions(quiz.questions);
     }
 
-    setQuestions(questions.map((question, index) => ({
-      ...question,
-      questionIndex: index + 1,
-    })));
+    // setQuestions(
+    //   questions.map((question, index) => ({
+    //     ...question,
+    //     questionIndex: index + 1,
+    //   }))
+    // );
```

Comment:
why comment these?

Reply:
oh wth, sorry, that was suppose to be uncommented, i was testing it. my mistake. 😥

Reply:
actually now i remembered, i have done it because, as we see, in the previous line, in if and else state, the state of the question have been set, but as the state will not update immediately, so the when the below line(which i have commented) get executed, the answer (i have shufled ) will not be there.
if you need questionIndex in each question, i can do this in another way, what do you say?

![image](https://github.com/wingkwong/react-quiz-component/assets/75330767/109efcd9-6f98-41e3-abd6-023c66a8e30c)

Reply:
![image](https://github.com/wingkwong/react-quiz-component/assets/75330767/ff3c3b84-63d7-4076-907e-b80dab8066f0)
i have still commented out the same line, and create some changes inside the if and else above the commented line.
is it okay?

Reply:
Oh ok. that one seems unnecessary. Your approach is fine.

Reply:
Thank you brother❤️